### ISSUE TITLE
No multiple server instances in V9 on OSX

### ISSUE THREAD

**Issue by Clintcinnati** [apohl79/audiogridder#230]
In v8 I usually run 6 server instances on my server imac and use it’s screens to interact with the plugin GUIs, with screen capture off altogether, and that server mac is connected to my DAW mac via single ethernet cable, all worked great and solid, only chore was each instance had the same ID number that I had to enter 1-6 on the respective servers.
Now with v9, I can only have one server instance because I get a warning about the port already being in use so the extra servers cannot start.
My workaround is using the v8 servers with the v9 plugins and it’s solid- no issue at all with six heavy plugins in maximum oversampling mode during over 7 hr session.

**Comment by apohl79** [comment]
You have to assign a different ID to each server instance. I’m wondering what’s different as this was always the case and has not changed. In beta9 the config locations changed, but AG is automatically migrating them. You can also use the “-id <num>” command line option. Either way, AG uses a different port once the ID is different.

**Comment by Clintcinnati** [comment]
I'm too bash illiterate to assign the "-id <num>" command line option, I'm not even sure how to start the servers via the command line.  So far using the above mentioned workaround is very stable (more stable than the Daw or OS sometimes!) I just wanted to see if the same thing happened with other people trying to run two or more beta 9 servers at once and getting that error (OSX 10.15.7).  I have tried the dumb way of trying to open the server settings window and changing the server ID in the provided box, but I can't get that to work with b9, yet it works in v8.  It's like I need to change the ID before starting the second server, but I can't do that without opening it, because it won't open (thus I realize the need to do it via command line, but I'd need a full hand-holding example of how to do that via the terminal with my limited terminal experience/memory)

Work-around notwithstanding, I'd like to thank you so very much for working on this, I'm in the midst of a project that is taxing my poor old mac for both screen area and processing, but with this server setup, both the screens for the plugins being open is solved while I can have both editor and mixer dedicated to their own respective screens.  
When I open and ensure the v8 servers are numbered and running with their own discreet ID numbers, then open my session on the Daw mac, all the settings for the plugins are there, and the connections are made without my intervention, which is absolutely great.   Running with no screen capture is a great way to use multiple macs, heck i have yet another big imac gathering dust, but for now I'll get this song mixed.
(too much coffee)

**Comment by apohl79** [comment]
Ok this sounds strange, it looks like you found some unintended use. I did a few tests with the last version yesterday, and it all works as expected. So, are you running all servers from the same user account? If so, I would be interested, how you get beta8 to run multiple different IDs without the commandline.

To use the terminal, here a quick guide:

1) open the Terminal app
2) run: `open /Applications/AudioGridderServer.app --args -id <number>`

**Comment by Clintcinnati** [comment]
Yes, all are running from same account on the mac.  I made duplicates of the app in the app folder so I have AudioGridder, AudioGridder copy, AudioGridder copy1, etc up to copy 5, for a total of 6.
I open all the copies, and they all open, yet have the same server ID when I open each of their Server Settings windows, so I manually change the number in each window from 1 to 6 and close each window.
On the plugin side I get a list six servers to connect to with the numbers 1-6 at the end of the local server computer name.  This way I can have six plugin instances in the daw, and have six windows for the plugins open at once on the server computer, that I control with a separate mouse and keyboard.  The responsiveness of the meters and controls of each plugin works vastly more smoothly than if I were to open on my daw computer.
Latency is compensated correctly, and once set up and I go through the numbering task of the servers at each startup of the server computer, everything just works.  Problems arise if I have the buffer set too low for some monster plugin, like Ozone 9, but even it works fine once the buffer is chosen correctly.

I think this is an excellent use, and it lets me use four monitors at one time with plugins being responsive.
The dream would be to have a server open each time a AudioGridder plugiin is opened in the DAW, or similar.  Screen capture bogged down my system so I tried turning it off alltogether, and I was pleased to find that controlling the plugin GUI on the server machine worked better than screen capturing and controling on the DAW host machine, for me anyway.
Thanks for the quick guide.
If I name the duplicate AudioGridder server applications with discreet names, is there a chance each instance will generate a separate config file (or whatever it is that saves the server ID) so the manual server ID assignment process will no longer be needed?  It is my theory that all instances share the same config saving only the ID number of the last number I entered and saved via the Server Settings pane.

**Comment by apohl79** [comment]
I suggest you start your server instances from a script and use the commandline interface. Alternatively you have to create multiple user accounts. AG does not allow you to choose different config files from the same user account.

**Comment by Clintcinnati** [comment]
I have set up multiple users on the server and tried spawning multiple server instances via the command line with b10 but same result, Only b8 server works with multiple instances under one user as I described previously.  The newest plugin works great with the b8 server though, so that’s good.   I run 7 server instances now, and sessions at 96k, no problems, all plugin gui are as responsive as they would be as if they were the only thing running.
As mentioned, I keep screen Capturing disabled and interact with the gui of each plugin at the server screen with separate mouse/kb.
Whatever happened between b8 and b9 to call up that port in use warning making this method of having multiple servers for me decreased the utility of the server, and I really have spent hours trying to get the newer server versions to work, starting by the command line, yet can not get this to work any other way than by using b8 and setting server id via the Server Settings window.
Was something added to later betas in a config file or something that I could change back to retain the way b8 worked?

I realize I should know more about BSD (mac) and be able to have multiple logins to I assume access multiple discreet config files for each corresponding server, but I don’t know how to get that to happen.

I’ll try to attach some pics of my rig.  
Oh the DAW is Harrison Mixbus 32c
![03E35430-0C5C-45AB-861B-B3C8EBC1CCFF](https://user-images.githubusercontent.com/74630181/101952458-b4558900-3bc6-11eb-8d2b-e4650ef9036b.jpeg)
![C92B6810-C157-4DF1-9EBE-EED505FC999A](https://user-images.githubusercontent.com/74630181/101952469-ba4b6a00-3bc6-11eb-86ad-a1df97efbd91.jpeg)
![6D25667B-13F7-42B0-BCE1-0D15E1D8222A](https://user-images.githubusercontent.com/74630181/101952475-bc152d80-3bc6-11eb-9bf2-817899e7c66f.jpeg)
![0AD351A9-B05D-4DD3-8F44-A7BE880A937C](https://user-images.githubusercontent.com/74630181/101952480-bddef100-3bc6-11eb-9284-3fe64d2443f5.jpeg)

**Comment by apohl79** [comment]
I guess it’s because AG is trying to kill a process running on the port it tries to bind to. This is likely preventing you from spawning another instance. 

I can make this optional and let the app ask in the first place. Or maybe try to pick the next available ID automatically.

For now: If the cmdline method does not work for you, you can manually change the ID in the config file before running the next instance. The config file path changed from beta8 to 9: https://audiogridder.com/server/#config

Hope that helps.

**Comment by Clintcinnati** [comment]
I’ll try changing the config file and see what happens.  I feel sure if running multiple servers from one account that I’m not the only one who would use that option-
these plugins take up lots of screen space so having their GUI stay on the server I find nice, and the bandwidth is freed for audio only it seems by the percentage indication per server in the connect-to list.
Thanks for this consideration.
-clint

**Comment by Clintcinnati** [comment]
By opening via the command line as you suggested I am now running b10 in multiple instances without problem.
The thing that was confusing me was when I provided an -id via the open command, the id did not show in the server settings window, nor was it saved to the config file when I closed the server.
However by opening each instance as you instructed with unique -id arguments, the servers opened without the “cannot bind” warning window, and are added to the list of servers shown on the plugin, with the id numbers as I had provided via the open command on the server mac.

And just like with b8, it’s working great!
